lib/ViewModels/CatalogItemNameSearchProviderViewModel.js

```diff
@@ -165,22 +166,33 @@ function findMatchingItemsRecursively(viewModel, searchInProgress, searchExpress
         }
 
         if (defined(item.items)) {
-            var loadPromise = item.load();
-            if (defined(loadPromise) && item.isLoading) {
-                var searchPromise = loadPromise.then(
-                    findItemsInLoadedGroup.bind(undefined, viewModel, searchInProgress, searchExpression, item, path.slice(), searchFilters)
-                    );
-                searchPromise = searchPromise.otherwise(ignoreLoadErrors);
-                promises.push(searchPromise);
-            } else {
-                findMatchingItemsRecursively(viewModel, searchInProgress, searchExpression, item, path, promises, searchFilters);
-            }
+            promises.push(loadAndSearchGroup(viewModel, item, searchInProgress, searchExpression, path, promises[0], searchFilters));
```

Comment:
So `promises[0]` is passed and waited for to avoid running every promise at once? Do you reckon you could explain that in a comment, took me a while to figure it out.